File: gherkin/dotnet/Gherkin/CucumberMessages/Types/Background.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Gherkin.CucumberMessages.Types
{
    public class Background
    {
        [DataMember(Name = "location")]
        public Location Location { get; set; }

        [DataMember(Name = "keyword")]
        public string Keyword { get; set; }

        [DataMember(Name = "name")]
        public string Name { get; set; }

        [DataMember(Name = "description")]
        public string Description { get; set; }

        [DataMember(Name = "steps")]
        public IReadOnlyCollection<Step> Steps { get; set; }

        [DataMember(Name = "id")]
        public string Id { get; set; }
    }
}

File: gherkin/dart/lib/src/ast/Background.dart
import 'package:gherkin/extensions.dart';
import 'package:gherkin/language.dart';

import 'Step.dart';
import 'StepsContainer.dart';

class Background extends StepsContainer
{
  static const empty = _InvalidBackground();

  const Background(Location location, String keyword, String name
      , String description , List<Step> steps)
    : super(location, keyword, name, description, steps);

  @override
  bool get isEmpty => false;

  @override
  bool get isNotEmpty => !isEmpty;
}

/// Convenience implementation of an invalid [Background] instance.
class _InvalidBackground
    extends Background
{
  const _InvalidBackground()
      : super(Location.empty, Strings.empty, Strings.empty, Strings.empty
        , const <Step>[] );

  @override
  bool get isEmpty => true;
}